### ISSUE TITLE
"Sleep Wake failure in EFI" on MacBookPro10,1

### ISSUE THREAD

**Issue by traviswparker** [dortania/Opencore-Legacy-Patcher#7]
After installing OpenCore, my 2012 rMBP 15" would crash if sleeping long enough to enter standby 
(AKA hibernate, determined by `pmset -a standbydelay`)

config.plist needs to support hibernation.
Under Misc/Boot/:
```
			<key>HibernateMode</key>
			<string>Auto</string>
```

The bare minimum config.plist that will boot this machine into Big Sur (identifying info removed):

```
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE plist PUBLIC "-//Apple//DTD PLIST 1.0//EN" "http://www.apple.com/DTDs/PropertyList-1.0.dtd">
<plist version="1.0">
<!--OpenCore 0.6.4 config-->
<dict>
	<key>Misc</key>
	<dict>
		<key>Boot</key>
		<dict>
			<key>HibernateMode</key>
			<string>Auto</string>
			<key>PickerMode</key>
			<string>Apple</string>
			<key>PollAppleHotKeys</key>
			<true/>
			<key>TakeoffDelay</key>
			<integer>0</integer>
			<key>Timeout</key>
			<integer>5</integer>
		</dict>
		<key>Security</key>
		<dict>
			<key>AllowNvramReset</key>
			<true/>
			<key>AllowSetDefault</key>
			<true/>
			<key>ApECID</key>
			<integer>0</integer>
			<key>AuthRestart</key>
			<false/>
			<key>BootProtect</key>
			<string>Bootstrap</string>
			<key>BlacklistAppleUpdate</key>
			<false/>
			<key>DmgLoading</key>
			<string>Signed</string>
			<key>EnablePassword</key>
			<false/>
			<key>ExposeSensitiveData</key>
			<integer>6</integer>
			<key>HaltLevel</key>
			<integer>2147483648</integer>
			<key>PasswordHash</key>
			<data></data>
			<key>PasswordSalt</key>
			<data></data>
			<key>ScanPolicy</key>
			<integer>0</integer>
			<key>SecureBootModel</key>
			<string>x86legacy</string>
			<key>Vault</key>
			<string>Optional</string>
		</dict>
		<key>Tools</key>
		<array>
			<dict>
				<key>Arguments</key>
				<string></string>
				<key>Auxiliary</key>
				<false/>
				<key>Name</key>
				<string>OpenShell.efi</string>
				<key>Comment</key>
				<string>OpenShell.efi</string>
				<key>Enabled</key>
				<true/>
				<key>Path</key>
				<string>OpenShell.efi</string>
				<key>RealPath</key>
				<false/>
				<key>TextMode</key>
				<false/>
			</dict>
		</array>
	</dict>
	<key>NVRAM</key>
	<dict>
		<key>Add</key>
		<dict>
			<key>4D1EDE05-38C7-4A6A-9CC6-4BCCA8B38C14</key>
			<dict>
				<key>DefaultBackgroundColor</key>
				<data>AAAAAA==</data>
				<key>UIScale</key>
				<data>Ag==</data>
			</dict>
			<key>4D1FDA02-38C7-4A6A-9CC6-4BCCA8B30102</key>
			<dict>
				<key>rtc-blacklist</key>
				<data></data>
			</dict>
			<key>7C436110-AB2A-4BBB-A880-FE41995C9F82</key>
			<dict>
				<key>boot-args</key>
				<string></string>
				<key>run-efi-updater</key>
				<string>No</string>
				<key>csr-active-config</key>
				<data>AAAAAA==</data>
			</dict>
		</dict>
		<key>Delete</key>
		<dict>
			<key>4D1EDE05-38C7-4A6A-9CC6-4BCCA8B38C14</key>
			<array>
				<string>UIScale</string>
				<string>DefaultBackgroundColor</string>
			</array>
			<key>4D1FDA02-38C7-4A6A-9CC6-4BCCA8B30102</key>
			<array>
				<string>rtc-blacklist</string>
			</array>
			<key>7C436110-AB2A-4BBB-A880-FE41995C9F82</key>
			<array>
				<string>boot-args</string>
			</array>
		</dict>
		<key>WriteFlash</key>
		<true/>
	</dict>
	<key>PlatformInfo</key>
	<dict>
		<key>Automatic</key>
		<true/>
		<key>CustomMemory</key>
		<false/>
		<key>Generic</key>
		<dict>
			<key>AdviseWindows</key>
			<false/>
			<key>SystemMemoryStatus</key>
			<string>Auto</string>
			<key>MLB</key>
			<string>...</string>
			<key>ProcessorType</key>
			<integer>0</integer>
			<key>ROM</key>
			<data>ESIzRFVm</data>
			<key>SpoofVendor</key>
			<true/>
			<key>SystemProductName</key>
			<string>MacBookPro11,2</string>
			<key>SystemSerialNumber</key>
			<string>...</string>
			<key>SystemUUID</key>
			<string>...</string>
		</dict>
		<key>UpdateDataHub</key>
		<true/>
		<key>UpdateNVRAM</key>
		<true/>
		<key>UpdateSMBIOS</key>
		<true/>
		<key>UpdateSMBIOSMode</key>
		<string>Create</string>
	</dict>
</dict>
</plist>
```

**Comment by khronokernel** [comment]
Ah thank you, will enable hibernate mode in the config. I meant to enable it a while ago however forgot

**Comment by TheMac6ine** [comment]
Hello, is that problem fixed? I used version 0.0.7 and when my MacBook 10.1 is to long in in standby, i think more than 24 hour it crashes. 


Sleep Wake failure in EFI

Failure code:: 0x00000000 0x0000001f

Please IGNORE the below stackshot

================================================================
Date/Time:        2020-12-10 06:16:09.275 +0100
OS Version:       ??? ??? (Build ???)
Architecture:     x86_64
Report Version:   32

Data Source:      Stackshots
Shared Cache:     EDB59884-BECA-35C6-A628-15DA943D22A5 slid base address 0x7fff200ba000, slide 0xba000

Event:            Sleep Wake Failure
Duration:         0.00s
Steps:            1

Boot args:        -v keepsyms=1 debug=0x100 agdpmod=pikera chunklist-security-epoch=0 -chunklist-no-rev2-dev

Time Awake Since Boot: 43s



Process:          swd [299]
Architecture:     x86_64
Footprint:        376 KB
Time Since Fork:  0s
Num samples:      1 (1)

  Thread 0x8ae    1 sample (1)    priority 4 (base 4)
  <thread QoS background (requested background), thread darwinbg, process darwinbg, IO tier 2>
  1  start + 1 (libdyld.dylib + 87601) [0x7fff203e3631] 1
    1  ??? [0x10ddf7454] 1
      1  ??? [0x10ddf71e8] 1
        1  __stack_snapshot_with_config + 10 (libsystem_kernel.dylib + 143386) [0x7fff203b601a] 1
         *1  ??? [0xffffff80002621f6] 1
           *1  ??? [0xffffff8000969ceb] 1
             *1  ??? [0xffffff8000873d11] 1
               *1  ??? [0xffffff80002851c9] 1
                 *1  ??? [0xffffff80002bbf0d] (running) 1

  Binary Images:
        0x7fff20393000 -     0x7fff203c1fff  libsystem_kernel.dylib (7195.50.7) <41068F5C-74E3-3C98-9256-6A18364FB5BA>  /usr/lib/system/libsystem_kernel.dylib
        0x7fff203ce000 -     0x7fff20408fff  libdyld.dylib (832.7.1)            <0C66AB9F-E22C-3286-B76B-DA4008698CD2>  /usr/lib/system/libdyld.dylib
Model: MacBookPro11,2, BootROM 427.0.0.0.0, 4 processors, Quad-Core Intel Core i7, 2,7 GHz, 16 GB, SMC 2.3f36
Graphics: kHW_IntelHD4000Item, Intel HD Graphics 4000, spdisplays_builtin
Graphics: kHW_NVidiaGeForceGT650MItem, NVIDIA GeForce GT 650M, spdisplays_pcie_device, 1 GB
Memory Module: BANK 0/DIMM0, 8 GB, DDR3, 1600 MHz, 0x80AD, 0x484D5434314753364D465238432D50422020
Memory Module: BANK 1/DIMM0, 8 GB, DDR3, 1600 MHz, 0x80AD, 0x484D5434314753364D465238432D50422020
AirPort: spairport_wireless_card_type_airport_extreme (0x14E4, 0x112), Broadcom BCM43xx 1.0 (7.77.111.1 AirPortDriverBrcmNIC-1669)
Bluetooth: Version 8.0.1f5, 3 services, 27 devices, 1 incoming serial ports
Network Service: Wi-Fi, AirPort, en0
Serial ATA Device: APPLE SSD SD512E, 500,28 GB
USB Device: USB 3.0 Bus
USB Device: USB 2.0 Bus
USB Device: Hub
USB Device: Hub
USB Device: Apple Internal Keyboard / Trackpad
USB Device: BRCM20702 Hub
USB Device: Bluetooth USB Host Controller
USB Device: USB 2.0 Bus
USB Device: Hub
USB Device: FaceTime HD Camera (Built-in)
Thunderbolt Bus: MacBook Pro, Apple Inc., 23.4
